app/src/main/java/ch/epfl/sdp/game/GameTimerFragment.java

```diff
@@ -0,0 +1,93 @@
+package ch.epfl.sdp.game;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.CountDownTimer;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import ch.epfl.sdp.R;
+import ch.epfl.sdp.databinding.FragmentGameTimerBinding;
+
+public class GameTimerFragment extends Fragment {
+
+    public interface GameTimeOutListener {
+        void onTimeOut();
+    }
+
+
+    private FragmentGameTimerBinding binding;
+
+    private static final String ARG_TIME = "time";
+    private static final int COUNTDOWN_INTERVAL = 1000;
+
+    private TextView textView;
+    public GameTimeOutListener listener;
+
+    public GameTimerFragment() {}
+
+    public static GameTimerFragment create(long time) {
+        GameTimerFragment fragment = new GameTimerFragment();
+        Bundle args = new Bundle();
+        args.putLong(ARG_TIME, time);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        binding = FragmentGameTimerBinding.inflate(inflater,container,false);
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        textView = binding.currentTime;
+
+        if (getArguments() != null) {
+            new CountDownTimer(getArguments().getLong(ARG_TIME), COUNTDOWN_INTERVAL) {
+                @Override
+                public void onTick(long millisUntilFinished) {
+                    textView.setText(millisToTimeString(millisUntilFinished));
+                }
+
+                @Override
+                public void onFinish() {
+                    if (listener != null) {
+                        listener.onTimeOut();
+                    }
+                    else {
+                        textView.setText(R.string.game_over);
+                    }
+                }
+            }.start();
+        } else {
+            textView.setText("--:--");
+        }
+    }
+
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        if (context instanceof GameTimeOutListener) {
+            this.listener = (GameTimeOutListener) context;
+        }
+    }
+
+    private String millisToTimeString(long millis) {
```

Comment:
should have tests

Reply:
I don't think one should test a private method as it is an implementation detail.

Reply:
Well there are different schools...
To stop the public vs private debate, I think this method is suitable for a public TimeHelper class.
This might, however be done in a future sprint :-)